cmake/Modules/SetGPUArchs.cmake

```diff
@@ -66,6 +66,13 @@ if(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
   evaluate_gpu_archs(CMAKE_CUDA_ARCHITECTURES)
 endif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
 
+# CMake architecture list entry of "80" means to build compute and sm. What we want is for the
+# newest arch only to build that way while the rest built only for sm.
+list(SORT CMAKE_CUDA_ARCHITECTURES ORDER ASCENDING)
+list(POP_BACK CMAKE_CUDA_ARCHITECTURES latest_arch)
+list(TRANSFORM CMAKE_CUDA_ARCHITECTURES APPEND "-real")
```

Comment:
Is `-real` a cmake command? What does this do?

Reply:
https://cmake.org/cmake/help/git-stage/prop_tgt/CUDA_ARCHITECTURES.html

Reply:
`-real` and `-virtual` are special keywords that can be used with `CMAKE_CUDA_ARCHITECTURES` to provide abstractions around different CUDA compilers code generation API.

For nvcc:

| input  | compiler invocation |
| ------------- | ------------- |
| 80  | --generate-code=arch=compute_80,code=[sm_80,compute_80]  |
| 80-virtual  | --generate-code=arch=compute_80,code=compute_80  |
| 80-real | --generate-code=arch=compute_80,code=sm_80 |

Reply:
Can I see the output of this command when `CMAKE_CUDA_ARCHITECTURES" is unset? 

We want SASS for all architectures we support, right? If we only include SASS ("-real"/) for 80, then users with anything but Ampere GPUs will experience looooong load/import times due to PTX-JIT to their present architecture. We do need to include PTX, but only for those who have GPUs we don't officially support (e.g. forward compatibility).

Reply:
I see, I had it backwards. The -real is appended to all but the last entry. I thought it was only being appended to the last entry. All good.

Reply:
> We want SASS for all architectures we support, right? If we only include SASS ("-real"/) for 80, then users with anything but Ampere GPUs will experience looooong load/import times 

You are correct. The code above is sneaky, as what we do is remove the 'newest' and only apply `-real` to any existing values. So input `70,80` becomes `70-real, 80` and input `80` becomes `80`